assets/js/modules/analytics/components/common/PropertySelectIncludingGA4.js

```diff
@@ -0,0 +1,133 @@
+import { Select, Option } from '../../../../material-components';
+import ProgressBar from '../../../../components/ProgressBar';
+import { MODULES_ANALYTICS_4, PROPERTY_CREATE } from '../../../analytics-4/datastore/constants';
+import { MODULES_ANALYTICS } from '../../datastore/constants';
+import { isValidAccountID } from '../../util';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function PropertySelectIncludingGA4() {
+	const accountID = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAccountID() );
+	const unmappedProperties = useSelect( ( select ) => select( MODULES_ANALYTICS ).getPropertiesIncludingGA4( accountID ) || [] );
+	const ga4PropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID() );
+	const uaPropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS ).getPropertyID() );
+	const isLoading = useSelect( ( select ) => (
+		! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' ) ||
+	! select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getProperties', [ accountID ] ) ||
+	! select( MODULES_ANALYTICS ).hasFinishedResolution( 'getProperties', [ accountID ] )
+	) );
+
+	const primaryPropertyType = useSelect( ( select ) => select( MODULES_ANALYTICS ).getPrimaryPropertyType() );
+
+	const ga4Dispatch = useDispatch( MODULES_ANALYTICS_4 );
+	const uaDispatch = useDispatch( MODULES_ANALYTICS );
+
+	const propertyID = primaryPropertyType === 'ga4' ? ga4PropertyID : uaPropertyID;
+
+	const onChange = useCallback( ( index, item ) => {
+		const newPropertyID = item.dataset.value;
+		const internalID = item.dataset.internalId; // eslint-disable-line sitekit/acronym-case
+		if ( propertyID !== newPropertyID ) {
+			trackEvent( 'analytics_setup', 'property_change', newPropertyID );
+			if ( newPropertyID.startsWith( 'UA-' ) ) {
+				uaDispatch.selectProperty( newPropertyID, internalID );
+				uaDispatch.setPrimaryPropertyType( 'ua' );
+
+				ga4Dispatch.setPropertyID( '' );
+				ga4Dispatch.setWebDataStreamID( '' );
+				ga4Dispatch.setMeasurementID( '' );
+			} else {
+				ga4Dispatch.selectProperty( newPropertyID );
+				uaDispatch.setPrimaryPropertyType( 'ga4' );
+
+				uaDispatch.setPropertyID( '' );
+				uaDispatch.setInternalWebPropertyID( '' );
+				uaDispatch.setProfileID( '' );
+			}
+		}
+	}, [ propertyID, ga4Dispatch, uaDispatch ] );
+
+	if ( ! isValidAccountID( accountID ) ) {
+		return null;
+	}
+
+	if ( isLoading ) {
+		return <ProgressBar small />;
+	}
+
+	const properties = unmappedProperties.map( ( p ) => {
+		if ( p?._id ) {
+			return {
+				...p,
+				// map GA4 properties to have fields like UA
+				id: p._id,
+				name: p.displayName,
+			};
+		}
+		return p;
+	} );
+
+	return (
+		<Select
+			className="googlesitekit-analytics__select-property"
+			label={ __( 'Property', 'google-site-kit' ) }
+			value={ propertyID }
+			onEnhancedChange={ onChange }
+			disabled={ ! isValidAccountID( accountID ) }
```

Comment:
This line doesn't make sense anymore because we already return `null` if the `accountID` is not valid. Let's get rid of this attribute then:

```suggestion
```